pandas/core/generic.py

```diff
@@ -5906,6 +5908,129 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _as_nullable_dtype(self: ABCSeries) -> ABCSeries:
+        """
+        Handle one Series
+
+        Rules:
+        If an object, see if we can infer string, boolean or integer, otherwise leave
+        alone
+        If an integer and not an extension type, convert to the Int64/Int32 type
+        (platform dependent)
+        If numeric, see if we can infer integer, otherwise try to use astype() to make
+        it integer.
+
+        """
+        dtype = self.dtype
+        new_dtype = dtype
+        changeit = False
+        constructit = True
+        result = self
+        target_int_dtype = "Int64"
+
+        if is_object_dtype(dtype):
+            new_dtype = lib.infer_dtype(self)
+            if new_dtype not in {"string", "boolean", "integer"}:
+                new_dtype = dtype
+            else:
+                changeit = True
+        elif is_integer_dtype(dtype):
+            if not is_extension_array_dtype(dtype):
+                new_dtype = "integer"
+                changeit = True
+        elif is_numeric_dtype(dtype):
+            new_dtype = dtype
+            try:
+                result = self.astype(target_int_dtype)
+                new_dtype = target_int_dtype
+                changeit = False
+                constructit = False
+            except TypeError:
+                pass
+
+        if changeit:
+            if new_dtype == "integer":
+                new_dtype = {
```

Comment:
This is confusing me a bit. Why are we making a new dict here? Can we do `registry.find(str(dtype).title()`?

Reply:
Yes, that will work.